### PR TITLE
rgw: rm non exists bucket index after use

### PR DESCRIPTION
**Pull request by imtzw**
if too many non exists bucket index there,
bucket list op will do too many io to check if
it still on disk, thus stuck.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by cbodley** [comment]
thanks @imtzw. can you please help us understand how you're reproducing these nonexistent index entries?

**Comment by imtzw** [comment]
> thanks @imtzw. can you please help us understand how you're reproducing these nonexistent index entries?

one occasion i know :
seq 1 8 | xargs -p 8 ${SEND_A_HTTP_DELETE_TO_RM_THE_SAME_OBJ}

other occasions may appear:
according to how "pending_map" works index remain "exists=false" when get removed but still have some active
reference.

File: src/cls/rgw/cls_rgw.cc [hunk 1]
```diff
@@ -1152,6 +1152,18 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
...
+	encode_obj_index_key(entry.key, &key_info);
+	CLS_LOG_BITX(bitx_inst, 20,
+		     "INFO: %s: removing map entry with key=%s exists=false",
+		     __func__, escape_str(key_info).c_str());
+	int ret = cls_cxx_map_remove_key(hctx, key_info);
+	if (ret < 0) {
+	  CLS_LOG_BITX(bitx_inst, 0, "ERROR: %s: unable to remove key, key=%s, exists=false, error=%d",
+		       __func__, escape_str(key_info).c_str(), ret); // dont return here
+	}
+    }
```
Comment (cbodley) [inline src/cls/rgw/cls_rgw.cc:1166]: i don't think this is the right place for this. the special cases for `CLS_RGW_OP_CANCEL` and `CLS_RGW_OP_DEL` below already handle this cleanup. if there are still bugs after https://github.com/ceph/ceph/pull/50041, we should fix that existing cleanup logic instead

Comment (imtzw) [reply src/cls/rgw/cls_rgw.cc:1166]: yes. this pr intends to repair the same issue as #50041.  No further bug found after it for now.

**Pull request closed without merging**